### DISCUSSION TITLE
404 error - external NPM

### DISCUSSION THREAD

**Discussion by hostrup** [llamafilm/tesla-http-proxy-addon#126] [General]
### The problem

I am trying to make this plugin work. 
I run home assistant via HAOS (i do it on a proxmox host ) but it's like it's own entity with own IP etc. 
To access it - i have another server - running the rest of my homelab - including 
NPM  (nginx proxy manager). 

From here i have lets encrypt working - and access to home assistant via my domain is working just fine. 

What i have done to make this work is: 

installed the addon. 
create a new entry i my npm config. 
here i have added 

tesla.mydomain.com 
which will proxy to 
10.0.0.6:4403 
i have tried both http and https to this port. 
the ip is my home assistant ip. 

i have created a static link 
so that the  /share/tesla is available at 
/www/tesla 

also i npm i have added this: 

location = /.well-known/appspecific/com.tesla.3p.public-key.pem {
    proxy_pass http://10.0.0.6:8123/local/tesla/com.tesla.3p.public-key.pem;
}

The URL works  - and the key.pem file is downloaded. 

i can run the addon - and i can generate all the tokens etc. 

But - from the log file it's not able to call the url i get this : 

HTTP/2 404 
server: openresty
date: Wed, 03 Jul 2024 11:08:42 GMT
content-type: text/plain; charset=utf-8
content-length: 14
referrer-policy: no-referrer
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 256 seconds. 6 retries 

Warning: left.
[13:12:30] werkzeug:INFO: 172.30.32.2 - - [03/Jul/2024 13:12:30] "GET / HTTP/1.1" 200 -

But when  i manally type in my domain in a browser like: 

https://$DOMAIN/.well-known/appspecific/com.tesla.3p.public-key.pem

it works just fine. 

is their some built in requirements for this to work? 

br Ronni

### What version of the addon are you using?

2.2.8 

### What version of Home Assistant Core are you using?

	core-2024.6.4

### What type of installation are you running?

Home Assistant OS

### What region of Tesla API are you using?

Europe, Middle East

### What model and year is your vehicle?

2022 model y

### What is your domain name (FQDN)?

tesla.hostrup.org

**Comment by llamafilm** [comment]
Moving to a discussion as this does not sound like a bug in the app.

I get an SSL error at https://tesla.hostrup.com/.well-known/appspecific/com.tesla.3p.public-key.pem
